zipline/testing/core.py

```diff
@@ -981,8 +981,19 @@ def wrapped(*args, **kwargs):
 
 
 class MockDailyBarReader(object):
+    def __init__(self, dates):
+        self.sessions = pd.DatetimeIndex(dates)
+
+    def load_raw_arrays(self, columns, start, stop, sids):
+        dates = self.sessions
+        output_dates = dates[(dates >= start) & (dates <= stop)]
```

Comment:
Should this barf if `start` or `stop` is outside the duration of `self.sessions`?